src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -1412,6 +1412,102 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                 writer.block_end();
             }
 
+            template <>
+            void GPU_Emitter::EMITTER_DECL(ngraph::op::BatchNorm)
+            {
+                const ngraph::op::BatchNorm* batchnorm =
+                    static_cast<const ngraph::op::BatchNorm*>(node);
+
+                auto& cudnn_emitter =
+                    external_function->get_primitive_emitter()->get_cudnn_emitter();
```

Comment:
My understanding is primitive, ctx etc is not part of externel_function, should be under backend. For example, multiple function can share primitive, ctx.

Reply:
At some point we should refactor the GPU backend to more align with the new backend API.